### PR TITLE
Test selinux with RedHat 8

### PR DESCRIPTION
**Pull request by francesco-giordano**
### Description of changes
* Test selinux with RedHat 8

### Tests
* Tested on docker and EC2

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cookbooks/aws-parallelcluster-install/recipes/disable_selinux.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+#
+# Cookbook:: aws-parallelcluster
+# Recipe:: awscli
```
Comment (eantonin) [inline cookbooks/aws-parallelcluster-install/recipes/disable_selinux.rb]: nitpick, non-blocking name of the recipe

File: test/common/libraries/os_properties.rb [hunk 2]
```diff
@@ -19,6 +19,14 @@ def redhat_ubi?
     virtualized? && inspec.os.name == 'redhat'
   end
 
+  def redhat?
```
Comment (eantonin) [inline test/common/libraries/os_properties.rb:22]: suggestion, non blocking: do you really need those? I think you can use `os.redhat?` and `os.centos?`

File: test/recipes/controls/aws_parallelcluster_install/disable_selinux.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+control 'disable_selinux' do
+  title 'Check if selinux is disabled'
+  describe selinux do
+    it { should be_disabled }
+    it { should_not be_enforcing }
+  end unless os_properties.redhat? || os_properties.centos? # Because it requires reboot of the instance
```
Comment (eantonin) [inline test/recipes/controls/aws_parallelcluster_install/disable_selinux.rb]: question, non blocking: are we sure only those ones needs reboot? see https://docs.chef.io/resources/selinux_state/#:~:text=NOTE%3A%20Switching%20to%20or%20from%20disabled%20requires%20a%20reboot!

File: cookbooks/aws-parallelcluster-install/recipes/disable_selinux.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+selinux_state "SELinux Disabled" do
+  action :disabled
+  only_if 'which getenforce'
```
Comment (eantonin) [inline cookbooks/aws-parallelcluster-install/recipes/disable_selinux.rb:21]: nitpick, non-blocking, can you explain with a comment why this is needed?

**Review by eantonin — DISMISSED** [review]

File: test/common/libraries/os_properties.rb (hunk 2 above)
Comment (francesco-giordano) [reply test/common/libraries/os_properties.rb:22]: `os.redhat` includes RedHat, centos and other OSes. The alternative is to use `os[:name] == 'redhat'` or `os.name == 'redhat'` directly

File: cookbooks/aws-parallelcluster-install/recipes/disable_selinux.rb (hunk 1 above)
Comment (francesco-giordano) [reply cookbooks/aws-parallelcluster-install/recipes/disable_selinux.rb]: Done

File: test/recipes/controls/aws_parallelcluster_install/disable_selinux.rb (hunk 3 above)
Comment (francesco-giordano) [reply test/recipes/controls/aws_parallelcluster_install/disable_selinux.rb]: If selinux would be active in other OSes, they would need the reboot as well. At the moment only Centos and Redhat are the OSes that require reboot.

**Review by eantonin — DISMISSED** [review]

**Review by enrico-usai — APPROVED** [review]

**Pull request merged**